variants/github_actions_ci/workflows/deploy_to_heroku.yml

```
@@ -0,0 +1,55 @@
+name: deploy_to_heroku
+on:
+  workflow_call:
+    inputs:
+      timeout_minutes:
+        type: number
+        required: false
+        default: 15
+      slack_icon:
+        type: string
+        required: false
+        default: https://github.com/ackama.png?size=48
+      slack_channel:
+        type: string
+        required: false
+      slack_username:
+        type: string
+        required: false
+        default: "Github Actions CI"
+    secrets:
+      heroku_api_key:
+        required: true
+      heroku_app_name:
+        required: true
+      heroku_email:
+        required: true
+      slack_webhook:
+        required: false
+jobs:
+  deploy_to_heroku:
+    runs-on: ubuntu-latest
+    timeout-minutes: ${{ inputs.timeout_minutes }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Deploy to Heroku Staging
+        uses: akhileshns/heroku-deploy@v3.8.9
+        with:
+          heroku_api_key: ${{ secrets.heroku_api_key }}
+          heroku_email: ${{ secrets.heroku_email }}
+          heroku_app_name: ${{ secrets.heroku_app_name }}
+      - name: Report to Slack
+        # 2022-09-16: Github Actions cannot currently reference `secrets` in an
+        # if expression. Workaround is to copy to secret to env var. In this
+        # case we already need the secret in an env var. Details:
+        # https://github.com/actions/runner/issues/520
+        if: "${{ env.SLACK_CHANNEL != '' }}"
```

🐘 we should be able to use `inputs` right? I think that'd be more consistent:

```suggestion
        if: "${{ inputs.SLACK_CHANNEL != '' }}"
```